subsys/bluetooth/audio/has.c

```diff
@@ -326,6 +326,35 @@ static int control_point_send(struct has_client *client, struct net_buf_simple *
 	return -ECANCELED;
 }
 
+static int control_point_send_all(struct net_buf_simple *buf)
+{
+	int result = 0;
+
+	for (size_t i = 0; i < ARRAY_SIZE(has_client_list); i++) {
+		struct has_client *client = &has_client_list[i];
+		int err;
+
+		if (!client->conn) {
+			continue;
+		}
+
+		if (!bt_gatt_is_subscribed(client->conn, PRESET_CONTROL_POINT_ATTR,
+					   BT_GATT_CCC_NOTIFY) &&
+		    !bt_gatt_is_subscribed(client->conn, PRESET_CONTROL_POINT_ATTR,
+					   BT_GATT_CCC_INDICATE)) {
```

Comment:
It isn't documented, but it looks like `bt_gatt_is_subscribed` can actually take a bitfield of `BT_GATT_CCC_NOTIFY` and `BT_GATT_CCC_INDICATE`: 

```c
		if (bt_conn_is_peer_addr_le(conn, cfg->id, &cfg->peer) &&
		    (ccc_value & ccc->cfg[i].value)) {
			return true;
		}
```

So this could be simplified as 
```suggestion
		if (!bt_gatt_is_subscribed(client->conn, PRESET_CONTROL_POINT_ATTR,
					   BT_GATT_CCC_NOTIFY | BT_GATT_CCC_INDICATE)) {
```

Reply:
But it may be risky as it isn't properly documented, so that may be unintentional in the function. Perhaps we should add that to the documentation of `bt_gatt_is_subscribed` such that the behavior can stay persistent?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/45326

Reply:
Done as you suggested